buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild/propagated-env-variables.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild
+
+import gradlebuild.basics.BuildEnvironment
+import org.gradle.api.tasks.testing.Test
+
+
+val propagatedEnvAllowList = listOf(
+    // Obviously necessary
+    "HOME",
+    "SHELL",
+    "TERM",
+
+    // Otherwise Windows complains "Unrecognized Windows Sockets error: 10106"
+    "SystemRoot",
+    "OS",
+
+    // For Android tests
+    "ANDROID_HOME",
+
+    // Used by Visual Studio
+    "USERNAME",
+    "USER",
+    "USERDOMAIN",
+    "USERPROFILE",
+    "LOCALAPPDATA",
+
+    // Used by Gradle test infrastructure
+    "REPO_MIRROR_URL",
+
+    // temp dir
+    "TMPDIR",
+    "TMP",
+    "TEMP",
+
+    // Seems important on Windows
+    "ALLUSERSPROFILE",
+    "PUBLIC",
+    "windir"
+)
+
+
+fun Test.configurePropagatedEnvVariables() {
+    if (BuildEnvironment.isCiServer) {
```

Comment:
❌ Local builds might also leak credentials. I think we should filter env vars for all builds.